Makefile

```
@@ -361,8 +361,17 @@ endif
 	fi;
 endif
 	
-	# Set rpath for libjulia-internal, which is moving from `../lib` to `../lib/julia`.  We only need to do this for Linux/FreeBSD
-ifneq (,$(findstring $(OS),Linux FreeBSD))
+	# Set rpath for libjulia-internal, which is moving from `../lib` to `../lib/julia`.
+ifeq ($(OS), Darwin)
+ifneq ($(DARWIN_FRAMEWORK),1)
+	install_name_tool -rpath '@loader_path/' '@loader_path/$(reverse_private_libdir_rel)/' $(DESTDIR)$(private_libdir)/libjulia-internal.$(SHLIB_EXT)
+	install_name_tool -rpath '@loader_path/julia/' '@loader_path/' $(DESTDIR)$(private_libdir)/libjulia-internal.$(SHLIB_EXT)
```

We can combine this into a single `install_name_tool` invocation:
```suggestion
	install_name_tool -rpath '@loader_path/' '@loader_path/$(reverse_private_libdir_rel)/' \
	                  -rpath '@loader_path/julia/' '@loader_path/' \
	                  $(DESTDIR)$(private_libdir)/libjulia-internal.$(SHLIB_EXT)
```

---

good to know, I'll make those changes

---

turns out that doesn't work if some of the paths are the same (I guess since the order is important): https://github.com/opensource-apple/cctools/blob/fdb4825f303fd5c0751be524babd32958181b3ed/misc/install_name_tool.c#L209

---

Ah, too bad.